iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/event/common/schema/PipeSchemaRegionWritePlanEvent.java

```diff
@@ -58,6 +60,26 @@ public PlanNode getPlanNode() {
     return planNode;
   }
 
+  public DeletionResource getDeletionResource() {
+    return deletionResource;
+  }
+
+  public void setDeletionResource(DeletionResource deletionResource) {
+    this.deletionResource = deletionResource;
+  }
+
+  @Override
+  public boolean internallyDecreaseResourceReferenceCount(String holderMessage) {
+    if (deletionResource != null) {
+      // Trigger hook function.
+      deletionResource.removeSelf();
```

Comment:
If a region group has 10 members, the leader will send the deletion to the follower 9 times and will generate 9 events. Is it OK for the first event to delete the resource after being sent?

Reply:
good catch! fixed